src/python/pants/backend/go/util_rules/sdk.py

```diff
@@ -125,9 +125,19 @@ async def setup_go_sdk_process(
         # TODO: Maybe could just use MAJOR.MINOR for version part here?
         "__PANTS_GO_SDK_CACHE_KEY": f"{goroot.version}/{goroot.goos}/{goroot.goarch}",
     }
+
     if request.replace_sandbox_root_in_args:
         env[GoSdkRunSetup.SANDBOX_ROOT_ENV] = "1"
 
+    # Disable the "coverage redesign" experiment on Go v1.20+ for now since Pants does not yet support it.
+    if goroot.is_compatible_version("1.20"):
+        exp_str = env.get("GOEXPERIMENT", "")
+        exp_fields = exp_str.split(",") if exp_str != "" else []
+        exp_fields = [exp for exp in exp_fields if exp != "coverageredesign"]
+        if "nocoverageredesign" not in exp_fields:
+            exp_fields.append("nocoverageredesign")
+        env["GOEXPERIMENT"] = ",".join(exp_fields)
```

Comment:
If we overrode someone's attempt at `coverageredesign` should we WARN of that fact?

Reply:
It could be worth it. Although if a user is informed enough to know about the coverage redesign `GOEXPERIMENT` and try to force enable it even though it is ordinarily enabled for `go`, then one could argue they should know better since they should be aware Pants doesn't support it yet.